### ISSUE TITLE
0.46.1 Princess Freeze Calculating Pathing

### ISSUE THREAD

**Issue by mcjomar** [MegaMek/megamek#1807]
0.46.1 MHQ Mergamek
Win10
 Java 1.8 I think?

### Description
Playing base defence as defender. AI is controlling Base Civilians, and CapCon Defence forces, and also FWL Attackers, Attackers #2 and Reinforcements (RAT tables don't seem to play ball with providing turrets to CapCon Base Turrets team).

Below bug occurred twice (two play attempts), usually during CapCon movement. Appears to be occurring with a hover SRM APC starting over water. Not sure what's happening, but CPU use is at 0% (occasionally 0.2%).
[megameklog.txt](https://github.com/MegaMek/megamek/files/4409240/megameklog.txt)
[borked.sav.gz](https://github.com/MegaMek/megamek/files/4409243/borked.sav.gz)

-----------------------------------------------------------------------------------
13:47:24,705 ERROR [megamek.common.pathfinder.LongestPathFinder] {Princess-precognition (Base Civilian Units)} 
Entity invalid. ID 15
java.lang.IllegalArgumentException: Entity invalid. ID 15
	at megamek.common.Compute.isPilotingSkillNeeded(Compute.java:483)
	at megamek.common.MovePath.addStep(MovePath.java:483)
	at megamek.common.MovePath.addStep(MovePath.java:257)
	at megamek.common.MovePath.addStep(MovePath.java:169)
	at megamek.common.pathfinder.MovePathFinder$NextStepsAdjacencyMap.getAdjacent(MovePathFinder.java:354)
	at megamek.common.pathfinder.MovePathFinder$NextStepsAdjacencyMap.getAdjacent(MovePathFinder.java:319)
	at megamek.common.pathfinder.AbstractPathFinder.run(AbstractPathFinder.java:304)
	at megamek.common.pathfinder.AbstractPathFinder.run(AbstractPathFinder.java:331)
	at megamek.client.bot.princess.PathEnumerator.recalculateMovesFor(PathEnumerator.java:268)
	at megamek.client.bot.princess.Precognition.run(Precognition.java:438)
	at java.lang.Thread.run(Unknown Source)

13:50:52,350 ERROR [megamek.common.pathfinder.LongestPathFinder] {Princess-precognition (Capellan Confederation)} 
Entity invalid. ID 15
java.lang.IllegalArgumentException: Entity invalid. ID 15
	at megamek.common.Compute.isPilotingSkillNeeded(Compute.java:483)
	at megamek.common.MovePath.addStep(MovePath.java:483)
	at megamek.common.MovePath.addStep(MovePath.java:257)
	at megamek.common.MovePath.addStep(MovePath.java:169)
	at megamek.common.pathfinder.MovePathFinder$NextStepsAdjacencyMap.getAdjacent(MovePathFinder.java:354)
	at megamek.common.pathfinder.MovePathFinder$NextStepsAdjacencyMap.getAdjacent(MovePathFinder.java:319)
	at megamek.common.pathfinder.AbstractPathFinder.run(AbstractPathFinder.java:304)
	at megamek.common.pathfinder.AbstractPathFinder.run(AbstractPathFinder.java:331)
	at megamek.client.bot.princess.PathEnumerator.recalculateMovesFor(PathEnumerator.java:268)
	at megamek.client.bot.princess.Precognition.run(Precognition.java:438)
	at java.lang.Thread.run(Unknown Source)

13:52:15,633 ERROR [megamek.common.pathfinder.LongestPathFinder] {Princess-precognition (Base Civilian Units)} 
Entity invalid. ID 15
java.lang.IllegalArgumentException: Entity invalid. ID 15
	at megamek.common.Compute.isPilotingSkillNeeded(Compute.java:483)
	at megamek.common.MovePath.addStep(MovePath.java:483)
	at megamek.common.MovePath.addStep(MovePath.java:257)
	at megamek.common.MovePath.addStep(MovePath.java:169)
	at megamek.common.pathfinder.MovePathFinder$NextStepsAdjacencyMap.getAdjacent(MovePathFinder.java:354)
	at megamek.common.pathfinder.MovePathFinder$NextStepsAdjacencyMap.getAdjacent(MovePathFinder.java:319)
	at megamek.common.pathfinder.AbstractPathFinder.run(AbstractPathFinder.java:304)
	at megamek.common.pathfinder.AbstractPathFinder.run(AbstractPathFinder.java:331)
	at megamek.client.bot.princess.PathEnumerator.recalculateMovesFor(PathEnumerator.java:268)
	at megamek.client.bot.princess.Precognition.run(Precognition.java:438)
	at java.lang.Thread.run(Unknown Source)

13:56:27,062 ERROR [megamek.common.pathfinder.LongestPathFinder] {Princess-precognition (Base Civilian Units)} 
Entity invalid. ID 15
java.lang.IllegalArgumentException: Entity invalid. ID 15
	at megamek.common.Compute.isPilotingSkillNeeded(Compute.java:483)
	at megamek.common.MovePath.addStep(MovePath.java:483)
	at megamek.common.MovePath.addStep(MovePath.java:257)
	at megamek.common.MovePath.addStep(MovePath.java:169)
	at megamek.common.pathfinder.MovePathFinder$NextStepsAdjacencyMap.getAdjacent(MovePathFinder.java:354)
	at megamek.common.pathfinder.MovePathFinder$NextStepsAdjacencyMap.getAdjacent(MovePathFinder.java:319)
	at megamek.common.pathfinder.AbstractPathFinder.run(AbstractPathFinder.java:304)
	at megamek.common.pathfinder.AbstractPathFinder.run(AbstractPathFinder.java:331)
	at megamek.client.bot.princess.PathEnumerator.recalculateMovesFor(PathEnumerator.java:268)
	at megamek.client.bot.princess.Precognition.run(Precognition.java:438)
	at java.lang.Thread.run(Unknown Source)

13:56:50,734 ERROR [megamek.common.pathfinder.LongestPathFinder] {Princess-precognition (Capellan Confederation)} 
Entity invalid. ID 15
java.lang.IllegalArgumentException: Entity invalid. ID 15
	at megamek.common.Compute.isPilotingSkillNeeded(Compute.java:483)
	at megamek.common.MovePath.addStep(MovePath.java:483)
	at megamek.common.MovePath.addStep(MovePath.java:257)
	at megamek.common.MovePath.addStep(MovePath.java:169)
	at megamek.common.pathfinder.MovePathFinder$NextStepsAdjacencyMap.getAdjacent(MovePathFinder.java:354)
	at megamek.common.pathfinder.MovePathFinder$NextStepsAdjacencyMap.getAdjacent(MovePathFinder.java:319)
	at megamek.common.pathfinder.AbstractPathFinder.run(AbstractPathFinder.java:304)
	at megamek.common.pathfinder.AbstractPathFinder.run(AbstractPathFinder.java:331)
	at megamek.client.bot.princess.PathEnumerator.recalculateMovesFor(PathEnumerator.java:268)
	at megamek.client.bot.princess.Precognition.run(Precognition.java:438)
	at java.lang.Thread.run(Unknown Source)

13:58:25,947 ERROR [megamek.common.pathfinder.LongestPathFinder] {Princess-precognition (Free Worlds League Force #2)} 
Entity invalid. ID 15
java.lang.IllegalArgumentException: Entity invalid. ID 15
	at megamek.common.Compute.isPilotingSkillNeeded(Compute.java:483)
	at megamek.common.MovePath.addStep(MovePath.java:483)
	at megamek.common.MovePath.addStep(MovePath.java:257)
	at megamek.common.MovePath.addStep(MovePath.java:169)
	at megamek.common.pathfinder.MovePathFinder$NextStepsAdjacencyMap.getAdjacent(MovePathFinder.java:351)
	at megamek.common.pathfinder.MovePathFinder$NextStepsAdjacencyMap.getAdjacent(MovePathFinder.java:319)
	at megamek.common.pathfinder.AbstractPathFinder.run(AbstractPathFinder.java:304)
	at megamek.common.pathfinder.AbstractPathFinder.run(AbstractPathFinder.java:331)
	at megamek.client.bot.princess.PathEnumerator.recalculateMovesFor(PathEnumerator.java:268)
	at megamek.client.bot.princess.Precognition.run(Precognition.java:438)
	at java.lang.Thread.run(Unknown Source)

**Comment by mcjomar** [comment]
[autosave.sav.gz](https://github.com/MegaMek/megamek/files/4409257/autosave.sav.gz)

also, have an autosave from probably the beginning of the turn?

**Comment by NickAragua** [comment]
Does kicking and replacing the bot fix the error?

**Comment by mcjomar** [comment]
I left the game (as I only tried playing for an hour during my WFH lunchbreak), so just tried loading the save (it crashed once after telling me I was disconnected).

Upon return (currently playing at time of reply) I discovered all players were ghosts (loading "borked" save file attached in thread).
I am replacing each AI Princessbot (with guesstimated behaviours for faction/disposition in the mission), and so far movement is occurring again.

It's a workaround I guess?
